tests/src/gcc/LaunchKernel.c

```diff
@@ -138,14 +138,43 @@ bool LaunchKernelArg3()
   return true;
 }
 
+bool LaunchKernelArg4()
+{
+  int A = 0;
+  int *A_d = NULL;
+  dim3 blocks       = {1,1,1};
+  dim3 threads      = {1,1,1};
+
+  // Allocate Device memory
+  HIPCHECK(hipMalloc((void**)&A_d, sizeof(int)));
+
+  char c = 1;
+  short s = 10;
+  int i = 100;
+  struct things t = {2,20,200};
```

Comment:
Don't we need to allocate these variables on device?

Reply:
Notice the kernel signature for kernel4, these are passed by value.